samples/example_vptr/example_vptr.cpp

```diff
@@ -0,0 +1,129 @@
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  Codeplay's ComputeCpp SDK
+ *
+ *  example_vptr.cpp
+ *
+ *  Description:
+ *    Sample code that walks through the basics of executing matrix addition
+ *    using the virtual pointer in SYCL.
+ *
+ **************************************************************************/
+
+#include <CL/sycl.hpp>
+
+#include <iostream>
+
+#include "pointer_alias.hpp"
+#include "virtual_ptr.hpp"
+
+using namespace cl::sycl;
+
+const size_t N = 100;
+const size_t M = 150;
+
+/* This sample creates three device-only arrays, then initialises them
+ * on the device. After that, it adds two of them together, storing the
+ * result in the third buffer. It then verifies the result of the kernel
+ * on the host by using a host accessor to gain access to the data. */
+int main() {
+  /* Destroying SYCL objects like queues and buffers blocks until all work
+   * associated with those objects is completed. */
+  {
+    queue myQueue;
+    cl::sycl::codeplay::PointerMapper pMap;
+
+    /* Create device-only 2D buffers of floats for the matrices. */
+    /*
+    buffer<float, 2> a(range<2>{N, M});
+    buffer<float, 2> b(range<2>{N, M});
+    buffer<float, 2> c(range<2>{N, M});
+    */
+    float* a = static_cast<float*>(SYCLmalloc(N * M * sizeof(float), pMap));
+    float* b = static_cast<float*>(SYCLmalloc(N * M * sizeof(float), pMap));
+    float* c = static_cast<float*>(SYCLmalloc(N * M * sizeof(float), pMap));
+
+    /* This kernel enqueue will initialise buffer a. The accessor "A" has
+     * write access. */
+    myQueue.submit([&](handler& cgh) {
+      auto A = pMap.get_access<access::mode::write,
+                               access::target::global_buffer, float>(a, cgh);
+      cgh.parallel_for<class init_a>(
+          range<1>(N * M), [=](id<1> index) { A[index] = index[0] * 2; });
+    });
+
+    /* This kernel enqueue will likewise initialise buffer b. The only
+     * accessor it specifies is a write accessor to b, so the runtime
+     * can use this information to recognise that these kernels are
+     * actually independent of each other. Therefore, they can be enqueued
+     * to the device with no dependencies between each other. */
+    myQueue.submit([&](handler& cgh) {
+      auto B = pMap.get_access<access::mode::write,
+                               access::target::global_buffer, float>(b, cgh);
+      cgh.parallel_for<class init_b>(
+          range<1>{N * M}, [=](id<1> index) { B[index] = index[0] * 2014; });
+    });
+
+    /* This kernel will actually perform the computation C = A * B. Since
```

Comment:
Should be `C = A + B`.

Reply:
Done. 0c2d66be8f9dee86bcaa729af98eca7ff31cbfa0